### DISCUSSION TITLE
SW is not working in HTTP.

### DISCUSSION THREAD

**Discussion by aravindkumarlotus** [jimmywarting/StreamSaver.js#274] [Q&A]
Hi, 
I'm trying to host a demo project with HTTP, but the Service Worker is not getting registered. if I host it in Https, the files are downloading but only accepting assets with HTTPS URLs, not HTTP. can you please advise how to move forward on this?

**Comment by jimmywarting** [comment]
the problem is that service worker is only able to be installed in secure contexts. eg localhost, 127.0.0.1 or 0.0.0.0 is also considered as secure cuz they are local so this is unfortunately not possible
